src/platform/scene/spawn-point.ts

```diff
@@ -0,0 +1,75 @@
+import { generateValidator, JSONSchema, ValidateFunction } from "../../validation";
+
+/** @alpha */
+export type SpawnPoint = {
+  name?: string;
+  position: {
+    x: number | number[];
+    y: number | number[];
+    z: number | number[];
+  };
```

Comment:
This is impossible to type on a statically typed language. Is it possible to change it to `position: MultiPosition | Vector3`?

Reply:
Yes, it's possible. But the example on the [docs](https://docs.decentraland.org/development-guide/scene-metadata/#spawn-location) does the following:
```
"position": {
        "x": [1,5],
        "y": 1,
        "z": [2,4]
}
```
I guess I don't fully understand why we should stop supporting that use case

cc @nearnshaw might have some context on how much this is used

Reply:
I'm not sure what `MultiPosition` type is, but we need to keep supporting positions like in that example

it's used quite a lot, and it's super useful
we use it in most of our scenes when we can

from the docs:
> You can set a whole region in the scene to act as a spawn point. By specifying an array of two numbers on any of the dimensions of the position, players will appear in a random location within this range of numbers. This helps prevent the overlapping of entering players.

Reply:
Committing to that schema is a very bad move from a protocolar point of view. It is literally unmarshable in many languages.

My proposal is:

```ts
type SpawnPoint = Either<SinglePosition | MultiPosition>
type SinglePosition = {
  x: number,
  y: number,
  z: number
}
type MultiPosition = {
  x: Array<number>,
  y: Array<number>,
  z: Array<number>
}
```

And then adjust the docs. What does it mean mixing arrays and numbers?
```
"position": {
        "x": [1,5],
        "y": 1, <<<<<<<????
        "z": [2,4]
}
```

Reply:
What about the many scenes that already have their spawn points defined like this?

Arrays and numbers are mixed because for example you often don't want the spawn _height_ to vary. 
 
 This equals to players spawning randomly anywhere in the area from x:1,y:1,z:2 to x:5,y:1,z:4
  "x": [1,5],  "y": 1,  "z": [2,4]
  
  I don't mind if from now on we'd have to write y as [1,1] for consistency
  But we do need to keep supporting this in existing scenes

Reply:
We already need to write an adapter on the CLI to remove the now invalid properties `policy` and `communication`. We would need to make that adapter transform from `"x": [1,5], "y": 1, "z": [2,4]` to `"x": [1,5], "y": [1,1], "z": [2,4]`

I'll go ahead and make the change then

cc @agusaldasoro